### PR TITLE
Ls 35149 config demo collector internal metrics adil

### PR DESCRIPTION
**Pull request by adil-sultan**
## What is the current behavior?
Currently we do not emit the collectors own metrics for tracking.

## What is the new behavior?
This PR creates a new pipeline which will send the collectors internal metrics to the public project in meta once the image is updated

---

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests(`make test`) for the changes have been added (for bug fixes / features) and pass
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Lint (`make lint`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by tratnayake — APPROVED** [review]
Appreciate the doc updates too!

File: config/collector-config.yml [hunk 1]
```diff
@@ -7,16 +7,25 @@ receivers:
...
-
+  prometheus:
+    config:
+      scrape_configs:
+        - job_name: 'otel-collector'
+          scrape_interval: 5s
+          static_configs:
+            - targets: ['0.0.0.0:8888']
+          relabel_configs:
+          - replacement: "internal-demo-otel-collector"
```
Comment (njvrzm) [inline config/collector-config.yml:18]: I think I'd rather make this `telemetry-generator` for consistency.

**Review by njvrzm — CHANGES_REQUESTED** [review]
Requesting a couple minor changes, but looks good!

**Review by njvrzm — APPROVED** [review]
🎉 thank you!

**Pull request merged**